File: src/main/kotlin/jimin/47week/입국심사.py
import sys
def solution(n, times):
    answer = sys.maxsize

    front = 1
    last = 1_000_000_000 * 1_000_000_000

    while front <= last:
        mid = (front + last) // 2
        num = 0
        for t in times:
            num += mid // t
        if num < n:
            front = mid + 1
        else:
            last = mid - 1
            answer = mid


    return answer

File: src/main/kotlin/hyunsoo/47week/입국심사.kt
package hyunsoo.`47week`

/**
 *
 * <문제>
 * [입국심사](https://school.programmers.co.kr/learn/courses/30/lessons/43238)
 *
 * - 아이디어
 *
 * - 트러블 슈팅
 *
 */
class `전현수_입국심사` {

    fun solution(n: Int, times: IntArray): Long {

        val sortedTimes = times.sortedBy { it }

        var answer: Long = 0

        var left = sortedTimes.first().toLong()
        var right = sortedTimes.last().toLong() * n

        while (left <= right) {

            val mid = (left + right) / 2
            var people: Long = 0

            sortedTimes.forEach { time ->
                people += mid / time
            }

            if (n <= people) {
                answer = mid
                right = mid - 1
            } else {
                left = mid + 1
            }

        }

        return answer
    }
}

fun main() {
    전현수_입국심사()
        .solution(6, intArrayOf(7, 10))
        .apply {
            println(this)
        }
}